core/src/main/java/com/linecorp/armeria/common/ContentDisposition.java

```diff
@@ -0,0 +1,426 @@
+                sb.append("\\\"");
+            } else {
+                sb.append(c);
+            }
+            escaped = !escaped && c == '\\';
+        }
+        // Remove backslash at the end..
+        if (escaped) {
+            sb.deleteCharAt(sb.length() - 1);
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Encodes the given header field param as describe in RFC 5987.
+     * @param input the header field param
+     * @param charset the charset of the header field param string,
+     *                only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported
+     * @return the encoded header field param
+     * @see <a href="https://tools.ietf.org/html/rfc5987">RFC 5987</a>
+     */
+    private static String encodeFilename(String input, Charset charset) {
+        final byte[] source = input.getBytes(charset);
+        final int len = source.length;
+        final StringBuilder sb = new StringBuilder(len << 1);
+        sb.append(charset.name());
+        sb.append("''");
+        for (byte b : source) {
+            if (isRFC5987AttrChar(b)) {
+                sb.append((char) b);
+            } else {
+                sb.append('%');
+                final char hex1 = Ascii.toUpperCase(Character.forDigit((b >> 4) & 0xF, 16));
+                final char hex2 = Ascii.toUpperCase(Character.forDigit(b & 0xF, 16));
+                sb.append(hex1);
+                sb.append(hex2);
+            }
+        }
+        return sb.toString();
+    }
+
+    @Override
+    public boolean equals(@Nullable Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (!(other instanceof ContentDisposition)) {
+            return false;
+        }
+        final ContentDisposition that = (ContentDisposition) other;
+        return type.equals(that.type) &&
+               Objects.equals(name, that.name) &&
+               Objects.equals(filename, that.filename) &&
+               Objects.equals(charset, that.charset);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(type, name, filename, charset);
+    }
+
+    /**
+     * Returns the header value for this content disposition as defined in RFC 6266.
+     * @see #parse(String)
+     */
+    public String asHeaderValue() {
+        if (strVal != null) {
+            return strVal;
+        }
+
+        final StringBuilder sb = TemporaryThreadLocals.get().stringBuilder();
+        sb.append(type);
+
+        if (name != null) {
+            sb.append("; name=\"");
+            sb.append(name).append('\"');
+        }
+        if (filename != null) {
+            if (charset == null || StandardCharsets.US_ASCII.equals(charset)) {
+                sb.append("; filename=\"");
+                sb.append(escapeQuotationsInFilename(filename)).append('\"');
+            } else {
+                sb.append("; filename*=");
+                sb.append(encodeFilename(filename, charset));
```

Comment:
Could we change `escapeQuotationsInFilename` and `encodeFilename` to accept `StringBuffer` for more efficiency?